src/app/src/test/kotlin/org/imperial/mrc/hint/integration/SessionTests.kt

```diff
@@ -0,0 +1,53 @@
+package org.imperial.mrc.hint.integration
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.assertj.core.api.AssertionsForClassTypes
+import org.imperial.mrc.hint.helpers.getTestEntity
+import org.imperial.mrc.hint.models.SessionFile
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.EnumSource
+import org.springframework.boot.test.web.client.postForEntity
+import org.springframework.http.HttpEntity
+import org.springframework.http.HttpHeaders
+import org.springframework.http.HttpStatus
+import org.springframework.http.MediaType
+
+class SessionTests : SecureIntegrationTests() {
+    @ParameterizedTest
+    @EnumSource(IsAuthorized::class)
+    fun `can set session files`(isAuthorized: IsAuthorized) {
+        //Get a valid hash
```

Comment:
So at the moment an invalid hash will throw a foreign key error? Might be a good idea to throw a custom 400 error with a user friendly message in case this does happen, not quite sure what, but indicating to the user that they've uploaded a corrupted file?